stl/inc/xstring

```diff
@@ -2290,8 +2294,10 @@ public:
     };
     _Bxty _Bx;
 
-    size_type _Mysize = 0; // current length of string
-    size_type _Myres  = 0; // current storage reserved for string
+    // invariant: _Myres >= _Mysize, and _Myres >= _SMALL_STRING_CAPACITY (after string's construction)
+    // both _Mysize and _Myres doesn't take account of the extra null terminator
+    size_type _Mysize = 0; // current length of string (size)
+    size_type _Myres  = 0; // current storage reserved for string (capacity)

```

Comment:
[won't be included in this pr] suggestion: we can make `_Myres` initialized to `_SMALL_STRING_CAPACITY` to make `_Myres >= _SMALL_STRING_CAPACITY` throughout the lifetime of `_String_val`.